doc/contributing.rst

```diff
@@ -14,11 +14,97 @@ If your contributions will make use of parallel backends for using more than
+
+Configuring git
+~~~~~~~~~~~~~~~
+
+Instructions for how to configure git can be found on 
+the git book `configuration <https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration>`_ page.
+
+Making your first pull request
+==============================
+Changes to hnn-core are typically made by "forking" the hnn-core
+repository, making changes to your fork (usually by "cloning" it to your
+personal computer, making the changes locally, and then "pushing" the local
+changes up to your fork on GitHub), and finally creating a "pull request" to incorporate
+your changes back into the shared "upstream" version of the codebase.
+
+In general you'll be working with three different copies of the hnn-core
+codebase: the official remote copy at https://github.com/jonescompneurolab/hnn-core
+(usually called ``upstream``), your remote fork of the upstream repository
+(similar URL, but with your username in place of ``hnn-core``, and usually
+called ``origin``), and the local copy of the codebase on your computer. The
+typical contribution process is to
+
+1. Make a fork of the `hnn-core <https://github.com/jonescompneurolab/hnn-core>`_
+   repository to your own account on github. Look for the Fork button in the top right corner
+
+2. On the terminal of your local computer clone the fork::
+
+    $ git clone https://github.com/<username>/hnn-core
+
+3. On the terminal of your local computer set up the remotes::
+
+    $ cd hnn-core
+    $ git remote add upstream https://github.com/jonescompneurolab/hnn-core
+
+4. Check that the remotes have been correctly added::
+
+    $ git remote -v
+
+   You should see::
+
+    origin	https://github.com/<username>/hnn-core (fetch)
+    origin	https://github.com/<username>/hnn-core (push)
+    upstream	https://github.com/jonescompneurolab/hnn-core (fetch)
+    upstream	https://github.com/jonescompneurolab/hnn-core (push)
+
+5. To start a new feature branch, we will copy the existing ``master`` branch from
+   the ``upstream`` remote and give it a specific name::
+
+    $ git fetch upstream master:cool_feature
+    $ git checkout cool_feature
+
+6. Make your changes relevant to the pull request
+
+7. To make a commit, you first have to add the files you have changed to the staging area::
+
+        $ git add -u
+
+   ensure they have been added correctly::
+
+        $ git status
+
+   make a commit::
+
+        $ git commit -m "your commit message"
+
+   and finally check that the commit has been added::
+
+        $ git log
+
+   Note: see the `numpy contributing guide <https://numpy.org/doc/stable/dev/development_workflow.html#writing-the-commit-message>`_
+   for tips on informative commit messages.
+
+8. Now push the feature branch to our fork::
+
+    $ git push origin cool_feature
+
+9. Go to https://github.com/jonescompneurolab/hnn-core/compare to open the pull request by selecting your feature branch.
+   You should also see this link on the terminal when you make the push. 
```

Comment:
Which link is supposed to show up on the terminal when you push? I'm not sure I follow what the point is here...

Reply:
I think it's just to draw attention to the fact that an automatic message gets generated about opening a PR? It's not exactly this link (the link check will fail on the PR otherwise) but they serve the same purpose